<launch>
    <!-- send pr2 urdf to param server -->
    <include file="$(find sot_pr2)/robots/upload_pr2.launch" />
    <!-- push robot_description to factory and spawn robot in gazebo -->
    <node name="spawn_pr2_model" pkg="gazebo" type="spawn_model" args="$(optenv ROBOT_INITIAL_POSE) -unpause -urdf -param robot_description -model pr2 -ros_namespace /gazebo" respawn="false" output="screen" />
    <!-- default bringup script -->
    <include file="$(find pr2_gazebo)/launch/pr2_bringup.launch" />

    <rosparam file="$(find pr2_controller_configuration_gazebo)/pr2_base_controller.yaml" command="load" />
    <rosparam file="$(find pr2_controller_configuration_gazebo)/pr2_odometry.yaml" command="load" />
    <rosparam file="$(find pr2_controller_configuration_gazebo)/pr2_gripper_controllers.yaml" command="load" />
    <node name="pr2_base_controllers_spawner" pkg="pr2_controller_manager" type="spawner" output="screen" args="--wait-for=/calibrated base_controller base_odometry r_gripper_controller l_gripper_controller" />

    <!-- Nodes on top of the controllers -->
    <group ns="r_gripper_controller">
        <node name="gripper_action_node" pkg="pr2_gripper_action" type="pr2_gripper_action">
            <!-- needed for stall detection in simulation with joint "jitter" -->
            <param name="stall_velocity_threshold" value="0.02" type="double"/>
            <param name="stall_timeout" value="0.5" type="double"/>
        </node>
    </group>
    <group ns="l_gripper_controller">
        <node name="gripper_action_node" pkg="pr2_gripper_action" type="pr2_gripper_action">
            <!-- needed for stall detection in simulation with joint "jitter" -->
            <param name="stall_velocity_threshold" value="0.02" type="double"/>
            <param name="stall_timeout" value="0.5" type="double"/>
        </node>
    </group>
</launch>
